pkg/build/builder/source_test.go

```diff
@@ -0,0 +1,26 @@
+package builder
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestCheckRemoteGit(t *testing.T) {
+	var err error
+	err = checkRemoteGit("https://github.com/openshift/private-repo.git", 10*time.Second)
+	switch v := err.(type) {
+	case gitAuthError:
+	default:
+		t.Errorf("expected gitAuthError, got %q", v)
+	}
+	err = checkRemoteGit("https://github.com/openshift/origin", 10*time.Second)
+	if err != nil {
+		t.Errorf("expected no error, got %q", err)
+	}
+	err = checkRemoteGit("https://254.254.254.254/foo/bar", 10*time.Second)
```

Comment:
I didn't say it would take 30 seconds. I said I didn't want it to take 30 seconds, which I don't :grin: